### PR TITLE
Add python lib for output production

### PR DESCRIPTION
**Pull request by mimir-d**
- run with `python3 scripts/demo.py`
- tests with `pytest -v`
- TODO.md, if not empty, has some questions/topics needing discussion (not merging until all those are done)

Note: PR is large, commits are organized to be read individually (atomic, mostly additive).

### CODE REVIEW

File: apis/python/tests/mocks.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    def decoded_obj(self, index: int) -> dict[str, JSON]:
+        """Decode an expected object from given output index"""
+        return json.loads(self.lines[index])
+
+
+def assert_json(line: str, expected: JSON):
+    actual = json.loads(line)
+
+    # remove timestamps, cant easily compare
+    if isinstance(expected, dict) and "timestamp" in actual:
```
Comment (dalwise) [inline apis/python/tests/mocks.py:22]:
```suggestion
    if isinstance(actual, dict) and "timestamp" in actual:
```

File: apis/python/ocptv/objects.py [hunk 2]
```diff
@@ -124,12 +107,181 @@ class File:
...
+    FIRMWARE = "FIRMWARE"
+    SYSTEM = "SYSTEM"
+    APPLICATION = "APPLICATION"
+
+
+@dc.dataclass
+class SoftwareInfo:
+    SPEC_OBJECT: ty.ClassVar[str] = "softwareInfo"
+
+    id: str = dc.field(
```
[the remark this answers is not in the document]
Comment (mimir-d) [reply apis/python/ocptv/objects.py:173]: for both id and type, i will argue: 1. these are internal only objects, no user is allowed to access them. 2. the id and type fields are always accessed with a getattr, so it's not a free variable that would overwrite the type/id funcs

File: apis/python/ocptv/objects.py [hunk 3]
```diff
@@ -124,12 +107,181 @@ class File:
...
+        metadata={"spec_field": "dutInfoId"},
+    )
+
+    name: ty.Optional[str] = dc.field(
+        metadata={"spec_field": "name"},
+    )
+
+    platform_infos: list[PlatformInfo] = dc.field(
```
[the remark this answers is not in the document]
Comment (mimir-d) [reply apis/python/ocptv/objects.py:270]: we can revert to older versions the typings; i know ive used 3.9 stuff, but it's easy to change and doesnt impact anything

File: apis/python/ocptv/formatter.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+def format_enum(variant: Enum) -> str:
+    """Format an enum by variant name"""
+    return "{}".format(variant.name)
+
+
+def format_timestamp(ts: float) -> str:
+    """Format an unix timestamp per spec"""
+    dt = datetime.fromtimestamp(ts)
+    return dt.astimezone().isoformat()
```
[the remark this answers is not in the document]
Comment (mimir-d) [reply apis/python/ocptv/formatter.py:13]: i was sure i was seeing iso8601 with tz dates in the samples output; i'll check

File: apis/python/ocptv/objects.py [hunk 5]
```diff
@@ -124,12 +107,181 @@ class File:
...
+    BUS = "BUS"
+    FUNCTION = "FUNCTION"
+    CONNECTOR = "CONNECTOR"
+
+
+@dc.dataclass
+class Subcomponent:
+    SPEC_OBJECT: ty.ClassVar[str] = "subcomponent"
+
+    type: ty.Optional[SubcomponentType] = dc.field(
```
[the remark this answers is not in the document]
Comment (mimir-d) [reply apis/python/ocptv/objects.py:123]: see my comment above

File: apis/python/ocptv/objects.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mimir-d) [reply apis/python/ocptv/objects.py:270]: ive reduced requirements to py37

File: apis/python/ocptv/formatter.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mimir-d) [reply apis/python/ocptv/formatter.py:13]: well, i checked the spec
> UTC time should be used but does not specify it as a requirement

and then there's an example showing `2022-01-05T06:37:41.211845017-06:00` as valid, which means local tz is also valid.

I'll add to the todo list to have some config block for users to be able to specify whether they want UTC or local tz.

**Comment by mimir-d** [comment]
left some items in TODO.md which will get addressed when moving to ocp-diag-python repo

**Review by dalwise — APPROVED** [review]

File: apis/python/ocptv/measurement.py [hunk 6]
```diff
@@ -29,7 +30,15 @@ def emit_impl(self, impl: MeasurementSeriesType):
 
 # Following object is a proxy type so we get future flexibility, avoiding the usage
 # of the low-level models.
+@export_api
 class Validator:
+    """
+    The `Validator` object represent a named validation that is relevant to a measurement or
```
[the remark this answers is not in the document]
Comment (mimir-d) [reply apis/python/ocptv/measurement.py:36]: will keep this one for todo; if i change anything now the approval gets reset

**Pull request merged**